util/topgen/merge.py

```diff
@@ -107,9 +108,14 @@ def elaborate_instance(instance, block: IpBlock):
         param_prefixes = ["Sec", "RndCnst", "MemSize"]
         name_top = cc_mod_name + param.name
         for prefix in param_prefixes:
-            if param.name.lower().startswith(prefix.lower()):
-                name_top = (prefix + cc_mod_name +
-                            param.name[len(prefix):])
+            if not param.name.startswith(prefix):
+                continue
+            else:
+                assert param.name != prefix
```

Comment:
We're not great at this in topgen, but could you make this a proper error (`ValueError`, maybe)? The point is that a failing assertion should mean a bug in the Python code, whereas this error condition can be triggered by an unfortunate choice of name in the hjson file.

Reply:
That's done now.